boot/bootutil/src/bootutil_misc.c

```diff
@@ -688,3 +696,28 @@ boot_set_confirmed(void)
     flash_area_close(fap);
     return rc;
 }
+
+/**
+ * Check if the version of the image is not older than required.
+ *
+ * @param req         Required minimal image version.
+ * @param ver         Version of the image to be checked.
+ *
+ * @return            0 if the version is sufficient, nonzero otherwise.
+ */
+int
+boot_is_version_sufficient(struct image_version *req,
```

Comment:
I'm not too familiar with this patch, but I think maybe this logic is wrong. The current logic will e.g. judge 2.0.0.0 to be older than 1.0.0.1.

Reply:
You're absolutely right... I'm going to update my change. Thank you.

Reply:
https://gist.github.com/d3zd3z/2567782363b3bf632eae6ee24b7187df should at least do the comparison in the logical manner, which isn't what semantic versions checks, but does compare them lexicographicaly.

However, I'm not sure that the build num is supposed to be included as part of the comparison, at least as this field was originally intended. Hopefully someone from MyNewt can chime in on the intent.